src/modules/course/schema/enrolledCourse.schema.ts

```diff
@@ -0,0 +1,50 @@
+import { Prop, Schema, SchemaFactory } from '@nestjs/mongoose';
+import { Document } from 'mongoose';
+import * as mongoose from 'mongoose';
+
+export type EnrolledCourseDocument = EnrolledCourse & Document;
+
+interface video {
+  num: number;
+  timestamp?: Date;
+}
+
+@Schema()
+export class EnrolledCourse {
+  @Prop({ default: [].fill(false, 10) })
+  videos_watched: boolean[];
+
+  @Prop({ default: [].fill(false, 10) })
+  Assignments_done: boolean[];
+
+  @Prop({ default: [] })
+  currentVideo: video[];
+
+  @Prop({ default: [] })
+  doubts: string[];
+
+  @Prop({})
+  Course: mongoose.Schema.Types.ObjectId;
+}
+
+export const EnrolledCourseSchema = SchemaFactory.createForClass(
+  EnrolledCourse,
+);
+
+EnrolledCourseSchema.pre('save', (next) => {
+  next();
+});
```

Comment:
Arrow func will not work, this is the working example:
```
EnrolledCourseSchema.pre('save', function (next) {

  var course = this;
  
  course['videos_watched'] = [true, true, false]

  next();
});

```

Reply:
Thanks.. async function and arrow function will not work so i will make the change